### ISSUE TITLE
Need to update MaskedInput widget

### ISSUE THREAD

**Issue by rahimov** [yiisoft/yii2#10850]
If use definitions, widget show error - "TypeError: $.inputmask is undefined".

Error show after composer update bower-asset/jquery.inputmask from 3.1.63 to 3.2.7 version.

Please change in \yii\widgets\MaskedInput lines 180-185
from:
```php
        if (is_array($this->definitions) && !empty($this->definitions)) {
            $js .= '$.extend($.' . self::PLUGIN_NAME . '.defaults.definitions, ' . Json::htmlEncode($this->definitions) . ');';
        }
        if (is_array($this->aliases) && !empty($this->aliases)) {
            $js .= '$.extend($.' . self::PLUGIN_NAME . '.defaults.aliases, ' . Json::htmlEncode($this->aliases) . ');';
        }
```

to:
```php
        if (is_array($this->definitions) && !empty($this->definitions)) {
            $js .= 'Inputmask.extendDefinitions(' . Json::htmlEncode($this->definitions) . ');';
        }
        if (is_array($this->aliases) && !empty($this->aliases)) {
            $js .= 'Inputmask.extendAliases(' . Json::htmlEncode($this->aliases) . ');';
        }
```

**Comment by wilsonfalai** [comment]
Before upgrading the yii inserted the following line in the beginning :
var inputmask_a2c3ba02 = {"alias":"date"};

Now insert the end of file

Translate
This has caused problems for me too :(

**Comment by rahimov** [comment]
@wilsonfalai 
change in line 149
from:
```php
$view->registerJs("var {$this->_hashVar} = {$encOptions};", View::POS_READY);
```
to:
```php
$view->registerJs("var {$this->_hashVar} = {$encOptions};", View::POS_HEAD);
```

**Comment by wilsonfalai** [comment]
Thaks! this resolved my problem :)

**Comment by cebe** [comment]
@samdark can you check whether this is related to #9783 (0bda5e362e5c38b08d042fa4d8859b72236036a1)

**Comment by klimov-paul** [comment]
Issue confirmed: it seems there is some BC break change in `inputmask` library somewhere between 3.1.63 and 3.2.7 versions.

**Comment by klimov-paul** [comment]
> can you check whether this is related to #9783 (0bda5e3)

It seems so. I have tries to stic "jquery.inputmask" version to "3.2.2", but it does not help: same error appears.

**Comment by klimov-paul** [comment]
Resolved by commit 2fe7ed05f0d8450bed0278077aeca5700de76606
